src/components/tx/confirmation-views/SwapOrder/index.tsx

```diff
@@ -0,0 +1,16 @@
+import SwapOrderConfirmation from '@/features/swap/components/SwapOrderConfirmationView'
+import type { AnySwapOrderConfirmationView } from '@safe-global/safe-gateway-typescript-sdk'
+import type { NarrowConfirmationViewProps } from '../types'
+
+interface SwapOrderProps extends NarrowConfirmationViewProps {}
+
+function SwapOrder({ txDetails, txInfo }: SwapOrderProps) {
+  return (
+    <SwapOrderConfirmation
+      order={txInfo as unknown as AnySwapOrderConfirmationView}
+      settlementContract={txDetails?.txData?.to?.value ?? ''}
+    />
+  )
+}
```

Comment:
I've checked the usage of `SwapOrder` and we only use it in places where we have guards for checking `txInfo`. Can't we type `txInfo` here to be more narrow (i.e. `Order`) instead of still resolving to `TransactionInfo`? That way we wouldn't need to cast.

This would match the structure of `StakingTxProps`

Reply:
Good point, fixed.